### PR TITLE
[BACKLOG-18996] PDI - Kafka Producer Step - Running a transformation with the Kafka Producer step in Spark or AEL produces a plugin class error

### PR DESCRIPTION
**Pull request by pavel-sakun**
Throw exception in case ClassLoadingPlugin doesn't have requested class registered.
@pentaho/ackbar @mkambol @DFieldFL

### CODE REVIEW

File: core/src/main/java/org/pentaho/di/core/plugins/PluginRegistry.java [hunk 1]
```diff
@@ -463,7 +463,14 @@ private String createSupplemantalKey( String pluginName, String id ) {
     }
 
     if ( plugin instanceof ClassLoadingPluginInterface ) {
-      return ( (ClassLoadingPluginInterface) plugin ).loadClass( pluginClass );
+      T aClass = ( (ClassLoadingPluginInterface) plugin ).loadClass( pluginClass );
+      if ( aClass == null ) {
+        throw new KettlePluginClassMapException( BaseMessages
+            .getString( PKG, "PluginRegistry.RuntimeError.NoValidClassRequested.PLUGINREGISTRY002", plugin.getName(),
+                pluginClass.getName() ) );
+      } else {
```
Comment (DFieldFL) [inline core/src/main/java/org/pentaho/di/core/plugins/PluginRegistry.java]: else is not needed here.

Comment (pavel-sakun) [reply core/src/main/java/org/pentaho/di/core/plugins/PluginRegistry.java]: Can't agree. Else branch is how it worked before (return plugin.loadClass), without it we're not getting services plugin provides.

Comment (DFieldFL) [reply core/src/main/java/org/pentaho/di/core/plugins/PluginRegistry.java]: Sorry didn't see the code below. Thanks 👍

**Comment by DFieldFL** [comment]
Looks like I can't merge...PluginRegistryUnitTest.java has a conflict. Would yo be able to fix this?

**Comment by pavel-sakun** [comment]
@DFieldFL done.

**Pull request merged**